services/121-service/src/transaction-job-processors/transaction-job-processors.service.ts

```diff
@@ -253,10 +252,22 @@ export class TransactionJobProcessorsService {
       }
     }
 
-    // 3. Start the transfer, save error transaction on failure and return early
-    let safaricomDoTransferResult: DoTransferReturnType;
+    // 3. Store 'waiting' transaction before starting transfer, because of callback.
+    const transaction = await this.createTransactionAndUpdateRegistration({
+      programId: input.programId,
+      paymentNumber: input.paymentNumber,
+      userId: input.userId,
+      calculatedTransferAmountInMajorUnit: input.transactionAmount,
+      financialServiceProviderId: financialServiceProvider.id,
+      registration,
+      oldRegistration,
+      isRetry: input.isRetry,
+      status: TransactionStatusEnum.waiting, // This will only go to 'success' via callback
+    });
+
+    // 4. Start the transfer, if failure update to error transaction and return early
     try {
-      safaricomDoTransferResult = await this.safaricomService.doTransfer({
+      await this.safaricomService.doTransfer({
         transactionAmount: input.transactionAmount,
```

Comment:
Update order to new order in DTO. See earlier comment.

Reply:
Updated!